manifests/db.pp

```diff
@@ -37,7 +37,6 @@
     password_hash => $hash,
     database      => $name,
     roles         => $roles,
-    require       => Mongodb_database[$name],
```

Comment:
If you're going to remove this dependency, could you add an autorequire for it? https://github.com/puppetlabs/puppetlabs-mongodb/blob/master/lib/puppet/type/mongodb_user.rb#L61-L65